packages/compass-schema/src/components/export-schema/export-schema-modal.tsx

```diff
@@ -0,0 +1,321 @@
+  maxHeight: '100%',
+  position: 'relative',
+});
+
+const exportPanelCodeAreaStyles = css({
+  position: 'relative',
+});
+
+const editorStyles = css({
+  width: spacing[6] * 10,
+  height: spacing[6] * 7,
+});
+
+const exportPanelCopyButtonStyles = css({
+  position: 'absolute',
+  top: spacing[2],
+  right: spacing[3],
+});
+
+const exportPanelSettingsAreaStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[2],
+});
+
+const ExportPanel: React.FunctionComponent<ExportPanelProps> = ({
+  settings,
+  exportInfo,
+}) => {
+  const handleCopy = useCallback(() => {
+    if (exportInfo.code) {
+      void navigator.clipboard.writeText(exportInfo.code);
+      return true;
+    }
+
+    return false;
+  }, [exportInfo.code]);
+
+  return (
+    <div className={exportPanelStyles}>
+      <div className={exportPanelCodeAreaStyles}>
+        {exportInfo.error ? (
+          <Banner variant={BannerVariant.Danger}>
+            Error: {exportInfo.error}
+          </Banner>
+        ) : (
+          <>
+            <WorkspaceContainer className={editorStyles}>
+              <JSONEditor
+                text={exportInfo.code || ''}
+                readOnly={true}
+                showFoldGutter={false}
+                showLineNumbers={false}
+              />
+            </WorkspaceContainer>
+            <div className={exportPanelCopyButtonStyles}>
+              <ActionButton
+                key="Copy"
+                label="Copy"
+                icon="Copy"
+                onClick={handleCopy}
+              />
+            </div>
+          </>
+        )}
+      </div>
+
+      {settings && (
+        <div className={exportPanelSettingsAreaStyles}>{settings}</div>
+      )}
+    </div>
+  );
+};
+
+function useSchemaConversion<T>(
+  schema: Schema,
+  settings: T,
+  converter: (schema: Schema, settings: T) => string
+): Export {
+  const exportInfo = useMemo(() => {
+    try {
+      return {
+        code: converter(schema, settings),
+      };
+    } catch (e) {
+      const errorMessage = (e as Error)?.message || '';
+      return { error: `Conversion failed: ${errorMessage}` };
+    }
+  }, [converter, schema, ...(settings && Object.values(settings))]);
+
+  return exportInfo;
+}
+
+const MongodbJsonSchema: React.FunctionComponent<{ schema: Schema }> = ({
+  schema,
+}) => {
+  const [includeId, setIncludeId] = useState<boolean>(true);
+  const [additionalProperties, setAdditionalProperties] =
+    useState<boolean>(true);
+  const [requireMandatoryProperties, setRequireMandatoryProperties] =
+    useState<boolean>(true);
+
+  const exportInfo: Export = useSchemaConversion(
+    schema,
+    { includeId, requireMandatoryProperties, additionalProperties },
+    (schema, settings) =>
+      JSON.stringify(exportMongodbJSONSchema(schema, settings), null, 2)
+  );
+
+  return (
+    <ExportPanel
+      exportInfo={exportInfo}
+      settings={
+        <>
+          <Checkbox
+            label="Include _id"
```

Comment:
Would it make sense to highlight `_id` as code here?

```suggestion
            label={<>Include <InlineCode>_id</InlineCode></>}
```

Reply:
(this applies for the other format as well)